pkg/query-service/app/inframetrics/hosts.go

```diff
@@ -0,0 +1,506 @@
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO(srikanthccv): only return resource attributes when we have a way to
+	// distinguish between resource attributes and other attributes.
+	filteredKeys := []v3.AttributeKey{}
+	for _, key := range attributeKeysResponse.AttributeKeys {
+		if slices.Contains(pointAttrsToIgnore, key.Key) {
+			continue
+		}
+		filteredKeys = append(filteredKeys, key)
+	}
+
+	return &v3.FilterAttributeKeyResponse{AttributeKeys: filteredKeys}, nil
+}
+
+func (h *HostsRepo) GetHostAttributeValues(ctx context.Context, req v3.FilterAttributeValueRequest) (*v3.FilterAttributeValueResponse, error) {
+	req.DataSource = v3.DataSourceMetrics
+	req.AggregateAttribute = "system_cpu_load_average_15m"
+	if req.Limit == 0 {
+		req.Limit = 50
+	}
+
+	attributeValuesResponse, err := h.reader.GetMetricAttributeValues(ctx, &req)
+	if err != nil {
+		return nil, err
+	}
+	if req.FilterAttributeKey != "host_name" {
+		return attributeValuesResponse, nil
+	}
+	hostNames := []string{}
+
+	for _, attributeValue := range attributeValuesResponse.StringAttributeValues {
+		if strings.Contains(attributeValue, "k8s-infra-otel-agent") {
+			continue
+		}
+		hostNames = append(hostNames, attributeValue)
+	}
+
+	req.FilterAttributeKey = "k8s_node_name"
+	req.DataSource = v3.DataSourceMetrics
+	req.AggregateAttribute = "system_cpu_load_average_15m"
+	if req.Limit == 0 {
+		req.Limit = 50
+	}
+
+	attributeValuesResponse, err = h.reader.GetMetricAttributeValues(ctx, &req)
+	if err != nil {
+		return nil, err
+	}
+	for _, attributeValue := range attributeValuesResponse.StringAttributeValues {
+		if strings.Contains(attributeValue, "k8s-infra-otel-agent") {
+			continue
+		}
+		hostNames = append(hostNames, attributeValue)
+	}
+
+	return &v3.FilterAttributeValueResponse{StringAttributeValues: hostNames}, nil
+}
+
+func getGroupKey(record model.HostListRecord, groupBy []v3.AttributeKey) string {
+	groupKey := ""
+	for _, key := range groupBy {
+		groupKey += fmt.Sprintf("%s=%s,", key.Key, record.Meta[key.Key])
+	}
+	return groupKey
+}
+
+func (h *HostsRepo) getMetadataAttributes(ctx context.Context,
+	req model.HostListRequest, hostNameAttrKey string) (map[string]map[string]string, error) {
+	hostAttrs := map[string]map[string]string{}
+
+	hasHostName := false
+	for _, key := range req.GroupBy {
+		if key.Key == hostNameAttrKey {
+			hasHostName = true
+		}
+	}
+
+	if !hasHostName {
+		req.GroupBy = append(req.GroupBy, v3.AttributeKey{Key: hostNameAttrKey})
+	}
+
+	mq := v3.BuilderQuery{
+		AggregateAttribute: v3.AttributeKey{
+			Key:      "system_cpu_load_average_15m",
+			DataType: v3.AttributeKeyDataTypeFloat64,
+		},
+		Temporality: v3.Unspecified,
+		GroupBy:     req.GroupBy,
+	}
+	query, err := helpers.PrepareTimeseriesFilterQuery(req.Start, req.End, &mq)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO(srikanthccv): remove this
+	query = strings.Replace(query, ".time_series_v4", ".distributed_time_series_v4", 1)
```

Comment:
Please add a comment here stating that. 

query can contain 
`time_series_v4, time_series_v4_6hrs, time_series_v4_1day` and you are adding distributed as a prefix to them, otherwise requires a lot of looking around